sdk/scheduler/src/main/java/org/apache/mesos/specification/DefaultServiceSpecification.java

```diff
@@ -29,19 +33,48 @@
     public static class Factory implements ConfigurationFactory<ServiceSpecification> {
 
         /**
-         * Call {@link DefaultServiceSpecification#getFactoryInstance()} instead.
+         * Subtypes to be registered by defaults. This list should include all
+         * {@link PlacementRule}s that are included in the library.
          */
-        private Factory() { }
+        private static final Collection<Class<?>> defaultRegisteredSubtypes = Arrays.asList(
+                AgentRule.class,
+                AndRule.class,
+                AttributeRule.class,
+                HostnameRule.class,
+                MaxPerAttributeRule.class,
+                NotRule.class,
+                OrRule.class,
+                TaskTypeRule.class);
```

Comment:
What are the requirements for a developer that wishes to write their own `PlacementRule` that doesn't make sense as an addition to the SDK?